cftime/_cftime.pyx

```diff
@@ -412,6 +461,189 @@ OverflowError in python datetime, probably because year < datetime.MINYEAR"""
+
+    **`only_use_python_datetimes`**: always return python datetime.datetime
+    objects and raise an error if this is not possible. Ignored unless
+    `only_use_cftime_datetimes=False`. Default `False`.
+
+    returns a datetime instance, or an array of datetime instances with
+    microsecond accuracy, if possible.
+
+    ***Note***: If only_use_cftime_datetimes=False and
+    use_only_python_datetimes=False, the datetime instances
+    returned are 'real' python datetime
+    objects if `calendar='proleptic_gregorian'`, or
+    `calendar='standard'` or `'gregorian'`
+    and the date is after the breakpoint between the Julian and
+    Gregorian calendars (1582-10-15). Otherwise, they are ctime.datetime
+    objects which support some but not all the methods of native python
+    datetime objects. The datetime instances
+    do not contain a time-zone offset, even if the specified `units`
+    contains one.
+    """
+    calendar = calendar.lower()
+    basedate = _dateparse(units)
+
+    can_use_python_datetime=\
+      ((calendar == 'proleptic_gregorian' and basedate.year >= MINYEAR) or \
+       (calendar in ['gregorian','standard'] and basedate > gregorian))
+    if not only_use_cftime_datetimes and only_use_python_datetimes:
+        if not can_use_python_datetime:
+            msg='illegal calendar or reference date for python datetime'
+            raise ValueError(msg)
+
+    unit, ignore = _datesplit(units)
+
+    # real-world calendars limited to positive reference years.
+    if calendar in ['julian', 'standard', 'gregorian', 'proleptic_gregorian']:
+        if basedate.year == 0:
+            msg='zero not allowed as a reference year, does not exist in Julian or Gregorian calendars'
+            raise ValueError(msg)
+
+    use_python_datetime = False
+    if only_use_python_datetimes and not only_use_cftime_datetimes:
+        # only_use_cftime_datetimes takes precendence
+        use_python_datetime = True
+    if not only_use_python_datetimes and not only_use_cftime_datetimes and can_use_python_datetime:
+        # if only_use_cftimes_datetimes and only_use_python_datetimes are False
+        # return python datetime if possible.
+        use_python_datetime = True
+
+    if use_python_datetime:
+        basedate = real_datetime(
+            basedate.year,
+            basedate.month,
+            basedate.day,
+            basedate.hour,
+            basedate.minute,
+            basedate.second,
+            basedate.microsecond
+        )
+    else:
+        basedate = to_calendar_specific_datetime(basedate, calendar)
+
+    if unit not in UNIT_CONVERSION_FACTORS:
+        raise ValueError("Unsupported time units provided, {!r}.".format(unit))
+
+    if unit in ["months", "month"] and calendar != "360_day":
+        raise ValueError("Units of months only valid for 360_day calendar.")
+
+    factor = UNIT_CONVERSION_FACTORS[unit]
+    scaled_times = factor * times
```

Comment:
Good idea to cast float `times` to `np.longdouble` prior to scaling.  I also added some code to `cast_to_int_if_safe` which prevents integer casting if the values are outside the longdouble integer range on the particular platform.  Let me know if that looks correct.

I make sure to cast any integer inputs to `int64` before doing any multiplication too.

> without casting to an array this will fail with list input.

Thanks for pointing this out -- I added a test case for list input and cast `times` as an array before doing anything else.